ql/src/java/org/apache/hadoop/hive/ql/QueryInfo.java

```diff
@@ -70,36 +80,57 @@ public String getExecutionEngine() {
     return executionEngine;
   }
 
-  public synchronized String getState() {
+  public String getState() {
     return state;
   }
 
+  /**
+   * The time the query began in milliseconds.
+   *
+   * @return The time the query began
+   */
   public long getBeginTime() {
-    return beginTime;
+    return TimeUnit.NANOSECONDS.toMillis(beginTime);
   }
 
-  public synchronized Long getEndTime() {
-    return endTime;
+  /**
+   * Get the end time in milliseconds. Only valid if {@link #isRunning()}
+   * returns false.
+   *
+   * @return Query end time
+   */
+  public long getEndTime() {
+    return TimeUnit.NANOSECONDS.toMillis(endTime);
   }
 
-  public synchronized void updateState(String state) {
+  public void updateState(String state) {
     this.state = state;
   }
 
   public String getOperationId() {
     return operationId;
   }
 
-  public synchronized void setEndTime() {
-    this.endTime = System.currentTimeMillis();
+  public void setEndTime() {
+    this.endTime = System.nanoTime();
   }
 
-  public synchronized void setRuntime(long runtime) {
-    this.runtime = runtime;
+  /**
+   * Set the amount of time the query spent actually running in milliseconds.
+   *
+   * @param runtime The amount of time this query spent running
+   */
+  public void setRuntime(long runtime) {
+    this.runtime = TimeUnit.MILLISECONDS.toNanos(runtime);
```

Comment:
the ```getRuntime``` returns value in milliseconds,  maybe there is no need to transfer the ```runtime``` back and forth?

Reply:
For simplicity sake, I wanted to keep all of the internal time values the same precision (nano):

```
 /*
  * Times are stored internally with nanosecond precision.
  */
```